apps/core/declarativewebcontainer.cpp

```diff
@@ -678,6 +678,7 @@ bool DeclarativeWebContainer::eventFilter(QObject *obj, QEvent *event)
 {
     if (obj == m_chromeWindow) {
         if (event->type() == QEvent::Close) {
+            emit applicationClosing();
```

Comment:
This should after line 691.

Reply:
Wanted to handle that tab closing as fast as possible before app exit but later seems to work just fine, fixed.